package/kedro_viz/data_access/repositories.py

```diff
@@ -0,0 +1,212 @@
+
+    def add(self, edge: GraphEdge):
+        self.edges_list.add(edge)
+
+    def as_list(self) -> List[GraphEdge]:
+        return list(self.edges_list)
+
+    def filter_by_node_ids(self, node_ids: Set[str]) -> List[GraphEdge]:
+        return [e for e in self.edges_list if {e.source, e.target}.issubset(node_ids)]
+
+
+class CatalogRepository:
+    _catalog: DataCatalog
+
+    def __init__(self):
+        self._layers_mapping = None
+
+    def get(self) -> DataCatalog:
+        return self._catalog
+
+    def set(self, value: DataCatalog):
+        self._catalog = value
+
+    @staticmethod
+    def strip_encoding(dataset_name: str) -> str:
+        return dataset_name.split("@")[0]
+
+    @property
+    def layers_mapping(self):
+        """Return layer mapping: dataset_full_name -> layer it belongs to in the catalog"""
+        if self._layers_mapping is not None:
+            return self._layers_mapping
+
+        if self._catalog.layers is None:
+            self._layers_mapping = {
+                self.strip_encoding(dataset_name): None
+                for dataset_name in self._catalog._data_sets
+            }
+        else:
+            self._layers_mapping = {}
+            for layer, dataset_names in self._catalog.layers.items():
+                self._layers_mapping.update(
+                    {
+                        self.strip_encoding(dataset_name): layer
+                        for dataset_name in dataset_names
+                    }
+                )
+        return self._layers_mapping
+
+    def get_dataset(self, dataset_name: str) -> Optional[AbstractDataSet]:
+        dataset_obj: Optional[AbstractDataSet]
+        if _KEDRO_VERSION.match(">=0.16.0"):
+            try:
+                dataset_obj = self._catalog._get_dataset(dataset_name)
+            except DataSetNotFoundError:
+                dataset_obj = None
+        else:
+            dataset_obj = self._catalog._data_sets.get(dataset_name)  # pragma: no cover
+        return dataset_obj
+
+    def get_layer_for_dataset(self, dataset_name: str) -> Optional[str]:
+        return self.layers_mapping.get(dataset_name)
+
+    @staticmethod
+    def is_dataset_param(dataset_name: str) -> bool:
+        """Returns whether a dataset is a parameter"""
+        return dataset_name.lower().startswith("param")
+
+
+class RegisteredPipelinesRepository:
+    def __init__(self):
+        self.pipelines_dict: Dict[str, RegisteredPipeline] = OrderedDict()
+        self.pipelines_node_ids_mapping: Dict[str, Set[str]] = defaultdict(set)
+
+    def add(self, pipeline_id: str):
+        self.pipelines_dict[pipeline_id] = RegisteredPipeline(id=pipeline_id)
+
+    def add_node(self, pipeline_id: str, node_id: str):
+        self.pipelines_node_ids_mapping[pipeline_id].add(node_id)
+
+    def get(self, pipeline_id: str) -> Optional[RegisteredPipeline]:
+        return self.pipelines_dict.get(pipeline_id)
+
+    def have(self, pipeline_id: str) -> bool:

```

Comment:
I'm not a huge fan of this method name.. I can't really explain why though but I'd prefer something like `has_pipeline` because it's clearer what you're checking this repository "has" if that makes sense.